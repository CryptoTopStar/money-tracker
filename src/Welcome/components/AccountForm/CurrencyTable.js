import React from 'react'
import CurrencyInput from '../CurrencyInput'
import { CURRENCY } from '../../constants'
import { Checkbox, Table } from 'semantic-ui-react'

const CurrencyTable = ({ currencies, balance, onChecked, updateBalance }) => (
  <Table basic="very" celled>
    <Table.Header>
      <Table.Row>
        <Table.HeaderCell style={{ width: '50%' }}>
          Currency
        </Table.HeaderCell>
        <Table.HeaderCell>Initial Balance</Table.HeaderCell>
      </Table.Row>
    </Table.Header>
    <Table.Body>
      {currencies.map(code => {
        return (
          <Table.Row key={code}>
            <Table.Cell>
              <Checkbox
                checked={balance[code] !== undefined}
                onChange={onChecked}
                value={code}
                label={CURRENCY[code].name}
              />
            </Table.Cell>
            <Table.Cell>
              <CurrencyInput
                value={balance[code]}
                code={code}
                update={updateBalance}
              />
            </Table.Cell>
          </Table.Row>
        )
      })}
    </Table.Body>
  </Table>
)

export default CurrencyTable
