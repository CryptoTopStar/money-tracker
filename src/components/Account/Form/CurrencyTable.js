import React from 'react'
import PropTypes from 'prop-types'
import { Checkbox, Input, Form } from 'semantic-ui-react'
import Amount from '../../../components/Amount'
import Currency from '../../../entities/Currency'

class CurrencyTable extends React.Component {
  componentWillMount() {
    this.updateCurrencyList(this.props.base, this.props.secondary)
    this.setInitialCurrencyCheckbox(this.props.base)
  }

  componentWillReceiveProps(nextProps) {
    this.updateCurrencyList(nextProps.base, nextProps.secondary)
    this.setInitialCurrencyCheckbox(nextProps.base)
  }

  updateCurrencyList(base, secondary) {
    this.currencies = [base, ...secondary]
  }

  setInitialCurrencyCheckbox(base) {
    if (this.props.form.currencies.length === 0) {
      this.props.toggleCurrency(base)
    }
  }

  handleCurrencyChange = (event, { value }) => {
    this.props.toggleCurrency(value)
  }

  handleBalanceChange = code => (event, { value }) => {
    this.props.changeBalance({ code: code, balance: value })
  }

  render() {
    return (
      <div>
        {this.currencies.map(code => (
          <Form.Group widths="equal" key={code}>
            <Form.Field className="checkbox">
              <Checkbox
                checked={this.props.form.currencies.includes(code)}
                onChange={this.handleCurrencyChange}
                value={code}
                label={Currency.name(code)}
              />
            </Form.Field>
            <Form.Field className="input-right">
              {this.props.form.id ? (
                <Amount
                  code={code}
                  value={Currency.toInt(
                    this.props.form.balance[code] || 0,
                    code
                  )}
                />
              ) : (
                <Input
                  labelPosition="right"
                  label={code}
                  type="number"
                  step={Currency.minAmount(code)}
                  autoComplete={false}
                  placeholder="Initial balance"
                  value={this.props.form.balance[code] || ''}
                  onChange={this.handleBalanceChange(code)}
                  fluid
                />
              )}
            </Form.Field>
          </Form.Group>
        ))}
      </div>
    )
  }
}

CurrencyTable.propTypes = {
  base: PropTypes.string.isRequired,
  secondary: PropTypes.arrayOf(PropTypes.string),
  form: PropTypes.shape({
    id: PropTypes.string,
    balance: PropTypes.objectOf(PropTypes.string),
    currencies: PropTypes.arrayOf(PropTypes.string)
  }).isRequired,
  toggleCurrency: PropTypes.func.isRequired,
  changeBalance: PropTypes.func.isRequired
}

export default CurrencyTable
