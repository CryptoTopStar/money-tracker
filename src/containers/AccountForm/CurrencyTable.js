import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Checkbox, Table } from 'semantic-ui-react'
import CurrencyInput from '../../components/CurrencyInput'
import { CURRENCY } from '../../constants/currency'
import {
  changeCurrencyCheckbox,
  changeCurrencyBalance
} from '../../actions/ui/accountForm'

class CurrencyTable extends React.Component {
  componentWillMount() {
    this.updateCurrencyList(this.props.base, this.props.secondary)
    this.setInitialBalanceCheckbox(this.props.base, this.props.balance)
  }

  componentWillReceiveProps(nextProps) {
    this.updateCurrencyList(nextProps.base, nextProps.secondary)
    this.setInitialBalanceCheckbox(nextProps.base, nextProps.balance)
  }

  updateCurrencyList(base, secondary) {
    this.currencies = [base, ...secondary]
  }

  setInitialBalanceCheckbox(base, balance) {
    if (Object.keys(balance).length === 0) {
      this.props.changeCurrencyCheckbox(base, true)
    }
  }

  handleCheckbox = (event, { value, checked }) => {
    this.props.changeCurrencyCheckbox(value, checked)
  }

  handleInput = code => event => {
    this.props.changeCurrencyBalance(code, event.target.rawValue)
  }

  render() {
    return (
      <Table basic="very" celled>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell style={{ width: '50%' }}>
              Currency
            </Table.HeaderCell>
            <Table.HeaderCell>Initial Balance</Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          {this.currencies.map(code => {
            return (
              <Table.Row key={code}>
                <Table.Cell>
                  <Checkbox
                    checked={this.props.balance[code] !== undefined}
                    onChange={this.handleCheckbox}
                    value={code}
                    label={CURRENCY[code].name}
                  />
                </Table.Cell>
                <Table.Cell>
                  <CurrencyInput
                    value={this.props.balance[code]}
                    code={code}
                    onChange={this.handleInput(code)}
                  />
                </Table.Cell>
              </Table.Row>
            )
          })}
        </Table.Body>
      </Table>
    )
  }
}

CurrencyTable.propTypes = {
  base: PropTypes.string,
  secondary: PropTypes.arrayOf(PropTypes.string),
  balance: PropTypes.objectOf(PropTypes.string),
  changeCurrencyCheckbox: PropTypes.func,
  changeCurrencyBalance: PropTypes.func
}

const mapStateToProps = state => ({
  base: state.settings.currency.base,
  secondary: state.settings.currency.secondary,
  balance: state.ui.accountForm.balance
})

export default connect(mapStateToProps, {
  changeCurrencyCheckbox,
  changeCurrencyBalance
})(CurrencyTable)
