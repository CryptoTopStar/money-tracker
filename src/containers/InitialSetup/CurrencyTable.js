import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Checkbox, Input, Form } from 'semantic-ui-react'
import {
  changeCurrencyCheckbox,
  changeCurrencyBalance
} from '../../actions/ui/accountForm'
import { getBaseCurrency, getSecondaryCurrency } from '../../selectors/currency'
import Currency from '../../entities/Currency'

class CurrencyTable extends React.Component {
  componentWillMount() {
    this.updateCurrencyList(this.props.base, this.props.secondary)
    this.setInitialBalanceCheckbox(this.props.base, this.props.balance)
  }

  componentWillReceiveProps(nextProps) {
    this.updateCurrencyList(nextProps.base, nextProps.secondary)
    this.setInitialBalanceCheckbox(nextProps.base, nextProps.balance)
  }

  updateCurrencyList(base, secondary) {
    this.currencies = [base, ...secondary]
  }

  setInitialBalanceCheckbox(base, balance) {
    if (Object.keys(balance).length === 0) {
      this.props.changeCurrencyCheckbox(base, true)
    }
  }

  handleCheckbox = (event, { value, checked }) => {
    this.props.changeCurrencyCheckbox(value, checked)
  }

  handleInput = code => (event, { value }) => {
    this.props.changeCurrencyBalance(code, value)
  }

  render() {
    return (
      <div>
        {this.currencies.map(code => {
          return (
            <Form.Group widths="equal" key={code}>
              <Form.Field style={{ paddingTop: '0.5em' }}>
                <Checkbox
                  checked={this.props.balance[code] !== undefined}
                  onChange={this.handleCheckbox}
                  value={code}
                  label={Currency.name(code)}
                />
              </Form.Field>
              <Form.Field className="input-right">
                <Input
                  labelPosition="right"
                  label={code}
                  type="number"
                  step="any"
                  autoComplete={false}
                  placeholder="Initial balance"
                  value={this.props.balance[code] || ''}
                  onChange={this.handleInput(code)}
                  fluid
                />
              </Form.Field>
            </Form.Group>
          )
        })}
      </div>
    )
  }
}

CurrencyTable.propTypes = {
  base: PropTypes.string,
  secondary: PropTypes.arrayOf(PropTypes.string),
  balance: PropTypes.objectOf(PropTypes.string),
  changeCurrencyCheckbox: PropTypes.func,
  changeCurrencyBalance: PropTypes.func
}

const mapStateToProps = state => ({
  base: getBaseCurrency(state),
  secondary: getSecondaryCurrency(state),
  balance: state.ui.accountForm.balance
})

export default connect(mapStateToProps, {
  changeCurrencyCheckbox,
  changeCurrencyBalance
})(CurrencyTable)
