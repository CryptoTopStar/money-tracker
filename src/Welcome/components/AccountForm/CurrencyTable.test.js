import React from 'react'
import renderer from 'react-test-renderer'
import CurrencyTable from './CurrencyTable'
global.navigator = {
  userAgent: 'node.js'
}

it('should render currency table with given currencies', () => {
  const tree = renderer
    .create(
      <CurrencyTable
        currencies={['USD', 'JPY']}
        balance={{ USD: 10, JPY: 200 }}
        onChecked={() => {}}
        update={() => {}}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
