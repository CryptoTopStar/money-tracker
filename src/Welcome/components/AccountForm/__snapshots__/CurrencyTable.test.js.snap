// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render currency table with given currencies 1`] = `
<table
  className="ui celled very basic table"
>
  <thead
    className=""
  >
    <tr
      className=""
    >
      <th
        className=""
        style={
          Object {
            "width": "50%",
          }
        }
      >
        Currency
      </th>
      <th
        className=""
      >
        Initial Balance
      </th>
    </tr>
  </thead>
  <tbody
    className=""
  >
    <tr
      className=""
    >
      <td
        className=""
      >
        <div
          className="ui checked checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={true}
            className="hidden"
            name={undefined}
            readOnly={true}
            tabIndex={0}
            type="checkbox"
            value="USD"
          />
          <label>
            US Dollar
          </label>
        </div>
      </td>
      <td
        className=""
      >
        <div
          className="ui fluid left labeled input"
        >
          <div
            className="ui label"
            onClick={[Function]}
          >
            $
          </div>
          <input
            data-cleave-ignore={
              Array [
                10,
                Object {
                  "initValue": 10,
                  "numeral": true,
                  "numeralDecimalScale": 2,
                },
                undefined,
                [Function],
                undefined,
                undefined,
              ]
            }
            name="USD"
            onChange={[Function]}
            onKeyDown={[Function]}
            type="text"
            value="10"
          />
        </div>
      </td>
    </tr>
    <tr
      className=""
    >
      <td
        className=""
      >
        <div
          className="ui checked checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={true}
            className="hidden"
            name={undefined}
            readOnly={true}
            tabIndex={0}
            type="checkbox"
            value="JPY"
          />
          <label>
            Japanese yen
          </label>
        </div>
      </td>
      <td
        className=""
      >
        <div
          className="ui fluid left labeled input"
        >
          <div
            className="ui label"
            onClick={[Function]}
          >
            Â¥
          </div>
          <input
            data-cleave-ignore={
              Array [
                200,
                Object {
                  "initValue": 200,
                  "numeral": true,
                  "numeralDecimalScale": 0,
                },
                undefined,
                [Function],
                undefined,
                undefined,
              ]
            }
            name="JPY"
            onChange={[Function]}
            onKeyDown={[Function]}
            type="text"
            value="200"
          />
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
